elastic/eui
# [Emotion] Convert final EuiDataGrid cell styles (Part 5)

## Summary

closes #6394 and #5685 😭 🥳 🎉 🎊 🍾 

This PR additionally includes some EuiDataGridHeaderCell cleanups from #7898 (see [individual commits](https://github.com/elastic/eui/pull/8013/commits) tagged with `[refactor]`), and some misc tech debt at the end of the commits.

Don't be scared by the files changed count, that's almost entirely loki VRT screenshots that I re-ran to update final differences. After this point, EuiDataGrid should be fully up to date and no screenshots should rerender differently locally 🎉 

## QA

- Row cell testing:
  - Go to https://eui.elastic.co/pr_8013/storybook/?path=/story/tabular-content-euidatagrid-rowheightsoptions-prop
  - [x] Smoke test that all cell height demos work as expected/are not broken
- Footer cell testing:
  - Go to https://eui.elastic.co/pr_8013/storybook/?path=/story/tabular-content-euidatagrid-gridstyle-prop--playground&globals=viewport:mobile2
  - [x] Confirm footer cells look/work as before
- Header cell testing
  -  Go to https://eui.elastic.co/pr_8013/storybook/?path=/story/tabular-content-euidatagrid--custom-header-content
  - [x] Confirm custom interactive children continues to shrink/truncate as expected and both the actions button and the sorting arrow do not shrink
  - Go to https://eui.elastic.co/pr_8013/storybook/?path=/story/tabular-content-euidatagrid--playground and confirm that the header cells look and work as before at the following display densities:
  - [x] Compact (the actions button should not make the header cell too tall)
  - [x] Expanded

### General checklist

- Browser QA
    - [x] Checked for **accessibility** including keyboard-only and ~screenreader modes~ reduced motion mode
    ~- [ ] Checked in **mobile**~
- Docs site QA - N/A
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
